### PR TITLE
[fix](inverted index) fix BE coredump because of not ignore case ensitivity for column name when create index

### PR DESCRIPTION
**Pull request by Tanya-W**
# Proposed changes

Issue Number: close #xxx

## Problem summary

fix BE coredump because of not ignore case ensitivity for column name when create index.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java [hunk 1]
```diff
@@ -34,6 +35,7 @@ public class IndexDef {
     private String indexName;
     private boolean ifNotExists;
     private List<String> columns;
+    private List<String> caseSensitivityColumns = Lists.newArrayList();
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java:38]: Please add some comments for the purpose.

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 2]
```diff
@@ -2166,7 +2167,8 @@ private boolean processAddIndex(CreateIndexClause alterClause, OlapTable olapTab
             }
         }
 
-        newIndexes.add(alterClause.getIndex());
+        alterIndex.setColumns(indexDef.getColumns());
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:2170]: Add some comments .

File: fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java (hunk 1 above)
Comment (Tanya-W) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java:40]: update

**Comment by Tanya-W** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Comment by Tanya-W** [comment]
run p0

**Pull request merged**